meteor/server/api/rest/rest.ts

```diff
@@ -104,25 +134,20 @@ function assignRoute(routeType: 'POST' | 'GET', resource: string, indexResource:
 			}
 			const result = fcn(p)
 
-			res.statusCode = 200
-			if (typeof result === 'object') {
-				res.setHeader('Content-Type', 'application/json')
-				res.end(JSON.stringify(result))
+			if (result && typeof result.then === 'function') {
```

Comment:
Just a thought: this could be an async function and just await the `fcn` and since it's already in a try catch no need to catch the error separately.

Reply:
Yeah that's a good hint and will make this much cleaner. Thanks!